graphqlapi/src/main/java/io/stargate/graphql/schema/fetchers/ddl/CreateIndexFetcher.java

```diff
@@ -0,0 +1,100 @@
+ * limitations under the License.
+ */
+package io.stargate.graphql.schema.fetchers.ddl;
+
+import com.google.common.base.Splitter;
+import graphql.schema.DataFetchingEnvironment;
+import io.stargate.auth.AuthenticationService;
+import io.stargate.auth.AuthorizationService;
+import io.stargate.db.datastore.DataStoreFactory;
+import io.stargate.db.query.Query;
+import io.stargate.db.query.builder.QueryBuilder;
+import io.stargate.db.schema.CollectionIndexingType;
+import io.stargate.db.schema.ImmutableCollectionIndexingType;
+import java.util.Arrays;
+import java.util.List;
+
+public class CreateIndexFetcher extends DdlQueryFetcher {
+
+  public CreateIndexFetcher(
+      AuthenticationService authenticationService,
+      AuthorizationService authorizationService,
+      DataStoreFactory dataStoreFactory) {
+    super(authenticationService, authorizationService, dataStoreFactory);
+  }
+
+  @Override
+  protected Query<?> buildQuery(
+      DataFetchingEnvironment dataFetchingEnvironment, QueryBuilder builder) {
+
+    String keyspaceName = dataFetchingEnvironment.getArgument("keyspaceName");
+    String tableName = dataFetchingEnvironment.getArgument("tableName");
+    String columnName = dataFetchingEnvironment.getArgument("columnName");
+
+    if (keyspaceName == null) {
+      List<String> parts = Splitter.on('.').splitToList(tableName);
+      if (parts.size() < 2) {
+        throw new IllegalArgumentException("Missing field argument keyspaceName @ 'createIndex'");
+      }
+      keyspaceName = parts.get(0);
+      tableName = parts.get(1);
+    }
+
+    String indexName =
+        dataFetchingEnvironment.getArgumentOrDefault(
+            "indexName", tableName + "_" + columnName + "_idx");
+
+    boolean ifNotExists =
+        dataFetchingEnvironment.getArgumentOrDefault("ifNotExists", Boolean.FALSE);
+    String customIndexClass =
+        dataFetchingEnvironment.getArgumentOrDefault("customIndexClass", null);
+
+    boolean indexEntries =
+        dataFetchingEnvironment.getArgumentOrDefault("indexEntries", Boolean.FALSE);
+    boolean indexKeys = dataFetchingEnvironment.getArgumentOrDefault("indexKeys", Boolean.FALSE);
+    boolean indexValues =
+        dataFetchingEnvironment.getArgumentOrDefault("indexValues", Boolean.FALSE);
+    boolean indexFull = dataFetchingEnvironment.getArgumentOrDefault("indexFull", Boolean.FALSE);
+
+    long paramsCount =
+        Arrays.asList(indexEntries, indexKeys, indexValues, indexFull).stream()
+            .filter(i -> i)
+            .count();
+
+    if (paramsCount > 1) {
+      throw new IllegalArgumentException(
+          "The indexEntries, indexKeys, indexValues and indexFull options are mutually exclusive");
+    }
+
+    CollectionIndexingType indexingType =
+        ImmutableCollectionIndexingType.builder()
+            .indexEntries(indexEntries)
+            .indexKeys(indexKeys)
+            .indexValues(indexValues)
+            .indexFull(indexFull)
+            .build();
+
```

Comment:
Will also need to call authorizationService here, similar to

https://github.com/stargate/stargate/blob/master/graphqlapi/src/main/java/io/stargate/graphql/schema/fetchers/ddl/TableFetcher.java